#Importing libraries and visualizing the DataFrame

In [57]:
import pandas as pd
from sklearn.preprocessing import Normalizer
from sklearn.cluster import KMeans
from sklearn import metrics

In [58]:
marketing ='/content/drive/MyDrive/marketing_campaign.csv'
df = pd.read_csv(marketing,sep='\t')

In [59]:
df

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
0      5524        1957  Graduation         Single  58138.0        0   
1      2174        1954  Graduation         Single  46344.0        1   
2      4141        1965  Graduation       Together  71613.0        0   
3      6182        1984  Graduation       Together  26646.0        1   
4      5324        1981         PhD        Married  58293.0        1   
...     ...         ...         ...            ...      ...      ...   
2235  10870        1967  Graduation        Married  61223.0        0   
2236   4001        1946         PhD       Together  64014.0        2   
2237   7270        1981  Graduation       Divorced  56981.0        0   
2238   8235        1956      Master       Together  69245.0        0   
2239   9405        1954         PhD        Married  52869.0        1   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
0            0  04-09-2012       58       635  ...                  7   
1            1  08-03-2014       38        11  ...                  5   
2            0  21-08-2013       26       426  ...                  4   
3            0  10-02-2014       26        11  ...                  6   
4            0  19-01-2014       94       173  ...                  5   
...        ...         ...      ...       ...  ...                ...   
2235         1  13-06-2013       46       709  ...                  5   
2236         1  10-06-2014       56       406  ...                  7   
2237         0  25-01-2014       91       908  ...                  6   
2238         1  24-01-2014        8       428  ...                  3   
2239         1  15-10-2012       40        84  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
0                0             0             0             0             0   
1                0             0             0             0             0   
2                0             0             0             0             0   
3                0             0             0             0             0   
4                0             0             0             0             0   
...            ...           ...           ...           ...           ...   
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
0            0              3         11         1  
1            0              3         11         0  
2            0              3         11         0  
3            0              3         11         0  
4            0              3         11         0  
...        ...            ...        ...       ...  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[2240 rows x 29 columns]

# Treating the DataFrame

**Reviewing all columns to determine their utility and potential elimination**

In [60]:
df.columns

Index(['ID', 'Year_Birth', 'Education', 'Marital_Status', 'Income', 'Kidhome',
       'Teenhome', 'Dt_Customer', 'Recency', 'MntWines', 'MntFruits',
       'MntMeatProducts', 'MntFishProducts', 'MntSweetProducts',
       'MntGoldProds', 'NumDealsPurchases', 'NumWebPurchases',
       'NumCatalogPurchases', 'NumStorePurchases', 'NumWebVisitsMonth',
       'AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5', 'AcceptedCmp1',
       'AcceptedCmp2', 'Complain', 'Z_CostContact', 'Z_Revenue', 'Response'],
      dtype='object')

In [61]:
df.drop(columns=['ID','Dt_Customer'],inplace=True)

**Performing a comprehensive check for null data and implementing necessary treatments**

In [81]:
df.isnull().sum()

Year_Birth                 0
Income                     0
Kidhome                    0
Teenhome                   0
Recency                    0
MntWines                   0
MntFruits                  0
MntMeatProducts            0
MntFishProducts            0
MntSweetProducts           0
MntGoldProds               0
NumDealsPurchases          0
NumWebPurchases            0
NumCatalogPurchases        0
NumStorePurchases          0
NumWebVisitsMonth          0
AcceptedCmp3               0
AcceptedCmp4               0
AcceptedCmp5               0
AcceptedCmp1               0
AcceptedCmp2               0
Complain                   0
Z_CostContact              0
Z_Revenue                  0
Response                   0
Education_2n Cycle         0
Education_Basic            0
Education_Graduation       0
Education_Master           0
Education_PhD              0
Marital_Status_Absurd      0
Marital_Status_Alone       0
Marital_Status_Divorced    0
Marital_Status_Married     0
Marital_Status


**Checking for rows with null data in the *Income* column**

In [63]:
nan_rows =  df[df['Income'].isna()]
nan_rows

Year_Birth   Education Marital_Status  Income  Kidhome  Teenhome  \
10          1983  Graduation        Married     NaN        1         0   
27          1986  Graduation         Single     NaN        1         0   
43          1959         PhD         Single     NaN        0         0   
48          1951  Graduation         Single     NaN        2         1   
58          1982  Graduation         Single     NaN        1         0   
71          1973    2n Cycle        Married     NaN        1         0   
90          1957         PhD        Married     NaN        2         1   
91          1957  Graduation         Single     NaN        1         1   
92          1973      Master       Together     NaN        0         0   
128         1961         PhD        Married     NaN        0         1   
133         1963  Graduation        Married     NaN        0         1   
312         1989  Graduation        Married     NaN        0         0   
319         1970  Graduation         Single     NaN        1         2   
1379        1970      Master       Together     NaN        0         1   
1382        1958  Graduation       Together     NaN        1         1   
1383        1964    2n Cycle         Single     NaN        1         1   
1386        1972         PhD       Together     NaN        1         0   
2059        1969      Master       Together     NaN        1         1   
2061        1981         PhD         Single     NaN        1         0   
2078        1971  Graduation        Married     NaN        1         1   
2079        1954      Master       Together     NaN        0         1   
2081        1955  Graduation         Single     NaN        0         1   
2084        1943      Master          Widow     NaN        0         0   
2228        1978    2n Cycle       Together     NaN        0         0   

      Recency  MntWines  MntFruits  MntMeatProducts  ...  NumWebVisitsMonth  \
10         11         5          5                6  ...                  7   
27         19         5          1                3  ...                  1   
43         80        81         11               50  ...                  2   
48         96        48          5               48  ...                  6   
58         57        11          3               22  ...                  6   
71         25        25          3               43  ...                  8   
90          4       230         42              192  ...                  9   
91         45         7          0                8  ...                  7   
92         87       445         37              359  ...                  1   
128        23       352          0               27  ...                  6   
133        96       231         65              196  ...                  4   
312        69       861        138              461  ...                  3   
319        67       738         20              172  ...                  7   
1379       39       187          5               65  ...                  5   
1382       87        19          4               12  ...                  5   
1383       49         5          1                9  ...                  7   
1386       17        25          1               13  ...                  7   
2059       52       375         42               48  ...                  3   
2061       82        23          0               15  ...                  6   
2078       82        71          1               16  ...                  8   
2079       83       161          0               22  ...                  6   
2081       95       264          0               21  ...                  7   
2084       75       532        126              490  ...                  1   
2228       53        32          2             1607  ...                  0   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
10               0             0             0             0             0   
27               0     

**Imputing the median of the *Income* column into its null values**

In [64]:
df['Income'].fillna(df['Income'].median(),inplace=True)

In [65]:
df.isnull().sum()

Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Complain               0
Z_CostContact          0
Z_Revenue              0
Response               0
dtype: int64

 **Transforming categorical variables into a numerical format suitable for further analysis**

In [66]:
df = pd.get_dummies(df,columns=['Education'],prefix=['Education'])
df = pd.get_dummies(df,columns=['Marital_Status'],prefix=['Marital_Status'])

In [67]:
df.head()

Year_Birth   Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0        1957  58138.0        0         0       58       635         88   
1        1954  46344.0        1         1       38        11          1   
2        1965  71613.0        0         0       26       426         49   
3        1984  26646.0        1         0       26        11          4   
4        1981  58293.0        1         0       94       173         43   

   MntMeatProducts  MntFishProducts  MntSweetProducts  ...  Education_Master  \
0              546              172                88  ...                 0   
1                6                2                 1  ...                 0   
2              127              111                21  ...                 0   
3               20               10                 3  ...                 0   
4              118               46                27  ...                 0   

   Education_PhD  Marital_Status_Absurd  Marital_Status_Alone  \
0              0                      0                     0   
1              0                      0                     0   
2              0                      0                     0   
3              0                      0                     0   
4              1                      0                     0   

   Marital_Status_Divorced  Marital_Status_Married  Marital_Status_Single  \
0                        0                       0                      1   
1                        0                       0                      1   
2                        0                       0                      0   
3                        0                       0                      0   
4                        0                       1                      0   

   Marital_Status_Together  Marital_Status_Widow  Marital_Status_YOLO  
0                        0                     0                    0  
1                        0                     0                    0  
2                        1                     0                    0  
3                        1                     0                    0  
4                        0                     0                    0  

[5 rows x 38 columns]

# Normalizing the data

**Normalizing the data to equalize the ranges**

In [68]:
values = Normalizer().fit_transform(df.values)

In [69]:
print(values)

[[3.36384675e-02 9.99322036e-01 0.00000000e+00 ... 0.00000000e+00
  0.00000000e+00 0.00000000e+00]
 [4.21255106e-02 9.99111905e-01 2.15586032e-05 ... 0.00000000e+00
  0.00000000e+00 0.00000000e+00]
 [2.74282525e-02 9.99602770e-01 0.00000000e+00 ... 1.39583982e-05
  0.00000000e+00 0.00000000e+00]
 ...
 [3.47402588e-02 9.99260316e-01 0.00000000e+00 ... 0.00000000e+00
  0.00000000e+00 0.00000000e+00]
 [2.82355542e-02 9.99576149e-01 0.00000000e+00 ... 1.44353549e-05
  0.00000000e+00 0.00000000e+00]
 [3.69339735e-02 9.99315375e-01 1.89017264e-05 ... 0.00000000e+00
  0.00000000e+00 0.00000000e+00]]


# Performing clustering and validating the clusters

* Silhoeutte -> O coeficiente de silhueta (Silhouette Score) é uma métrica que varia de -1 a 1, onde valores mais próximos de 1 indicam que as amostras estão bem agrupadas, enquanto valores próximos de -1 indicam que as amostras estão mal agrupadas. Um valor positivo do coeficiente de silhueta geralmente é considerado bom, indicando que os clusters são bem definidos e distintos uns dos outros.

* Davies Bouldin -> O índice Davies-Bouldin é uma medida de similaridade entre os clusters. Quanto mais próximo de zero o valor, melhor a partição dos clusters é considerada. Ele é calculado como a média das similaridades intracluster dividida pelas maiores dissimilaridades intercluster. Portanto, um valor mais próximo de zero indica que os clusters estão compactos e bem separados.


* Calinski -> O índice Calinski-Harabasz é uma medida de quão densos e bem separados os clusters estão. Quanto maior o valor do índice, melhor é a clusterização. Ele é calculado como a razão da dispersão entre clusters e a dispersão dentro dos clusters, multiplicado pelo fator de correção. Portanto, um valor maior indica uma melhor separação entre os clusters.

***
***

* Silhouette Score: The silhouette coefficient is a metric that ranges from -1 to 1, where values closer to 1 indicate that samples are well-clustered, while values close to -1 indicate that samples are poorly clustered. A positive silhouette coefficient is generally considered good, indicating that clusters are well-defined and distinct from each other.

* Davies-Bouldin Index: The Davies-Bouldin index is a measure of similarity between clusters. The closer the value is to zero, the better the clustering partition is considered. It is calculated as the average of intracluster similarities divided by the largest intercluster dissimilarities. Therefore, a value closer to zero indicates that clusters are compact and well-separated.

* Calinski-Harabasz Index: The Calinski-Harabasz index is a measure of how dense and well-separated clusters are. The higher the index value, the better the clustering. It is calculated as the ratio of between-cluster dispersion and within-cluster dispersion, multiplied by the correction factor. Therefore, a higher value indicates better separation between clusters.

In [70]:
def clustering_algorithm (n_clusters, dataset):
  Kmeans = KMeans(n_clusters=n_clusters,n_init=10,max_iter=300)
  labels = Kmeans.fit_predict(dataset)
  s = metrics.silhouette_score(dataset,labels,metric='euclidean')
  dbs = metrics.davies_bouldin_score(dataset,labels)
  c = metrics.calinski_harabasz_score(dataset,labels)
  return s, dbs, c

In [71]:
s1, dbs1, c1 = clustering_algorithm(3,values)
print(f'Silhouette-> {s1}\nDavies Bouldin-> {dbs1}\nCalinski->{c1}')

Silhouette-> 0.6373260467253079
Davies Bouldin-> 0.5770273764353401
Calinski->2484.835695521793


In [72]:
s2,dbs2,c2 = clustering_algorithm(5,values)
print(f'Silhouette-> {s2}\nDavies Bouldin-> {dbs2}\nCalinski->{c2}')

Silhouette-> 0.5489096078580814
Davies Bouldin-> 0.5956463531490014
Calinski->3644.7829537453767


In [73]:
s3, dbs3, c3 = clustering_algorithm(10,values)
print(f'Silhouette-> {s3}\nDavies Bouldin-> {dbs3}\nCalinski->{c3}')

Silhouette-> 0.43798093151544015
Davies Bouldin-> 0.5121625320581057
Calinski->6994.896348473821


In [74]:
s4, dbs4, c4 = clustering_algorithm(20,values)
print(f'Silhouette-> {s4}\nDavies Bouldin-> {dbs4}\nCalinski->{c4}')

Silhouette-> 0.3135029650431013
Davies Bouldin-> 0.689633045041189
Calinski->9032.86161921743


Após uma análise abrangente, optamos por utilizar a clusterização com três clusters, pois é onde obtivemos o melhor resultado global e um equilíbrio significativo entre as métricas avaliadas.

Aqui estão os resultados das métricas de avaliação para esta configuração específica:

Silhouette Score: 0.5489
Davies-Bouldin Index: 0.5956
Calinski-Harabasz Score: 3644.783


Destaca-se que nesta configuração de clusterização, o Silhouette Score retorna o valor mais próximo de 1, indicando uma boa estruturação dos clusters. O Davies-Bouldin Index apresenta o segundo menor valor entre as configurações avaliadas, sugerindo uma boa separação e coesão dos clusters. Embora o Calinski-Harabasz Score registre um resultado relativamente mais baixo nesta configuração específica, optamos por sua utilização devido à sua contribuição para um equilíbrio geral na estrutura dos clusters.

É importante ressaltar que daremos maior ênfase à métrica Silhouette Score, dada sua popularidade e confiabilidade.

***
***

After a comprehensive analysis, we have opted to utilize clustering with three clusters as it yielded the best overall result and a significant balance among the evaluated metrics.

Here are the results of the evaluation metrics for this specific configuration:

Silhouette Score: 0.5489
Davies-Bouldin Index: 0.5956
Calinski-Harabasz Score: 3644.783
It is noteworthy that in this clustering configuration, the Silhouette Score returns the value closest to 1, indicating a well-structured clustering. The Davies-Bouldin Index presents the second lowest value among the evaluated configurations, suggesting good separation and cohesion of the clusters. Although the Calinski-Harabasz Score registers a relatively lower result in this specific configuration, we chose to use it due to its contribution to an overall balance in the cluster structure.

It is important to emphasize that we will give greater emphasis to the Silhouette Score metric, given its popularity and reliability.


# Understanding the attribute values within the cluster

**Performing clustering and creating a column in the DataFrame for these clusters**

In [75]:
Kmeans = KMeans(n_clusters=3,n_init=10,max_iter=300)
y_pred = Kmeans.fit_predict(values)
labels = Kmeans.labels_
df['cluster'] = labels

In [76]:
df.groupby('cluster').describe()

Year_Birth                                                          \
             count         mean        std     min     25%     50%     75%   
cluster                                                                      
0            418.0  1974.571770  10.342614  1946.0  1969.0  1976.0  1983.0   
1           1795.0  1967.355432  11.905500  1893.0  1958.0  1968.0  1976.0   
2             27.0  1975.962963  11.791397  1950.0  1970.0  1976.0  1985.5   

                 Income                ... Marital_Status_Widow       \
            max   count          mean  ...                  75%  max   
cluster                                ...                             
0        1996.0   418.0  24142.803828  ...                  0.0  1.0   
1        1995.0  1795.0  59467.324234  ...                  0.0  1.0   
2        1996.0    27.0   6575.037037  ...                  0.0  0.0   

        Marital_Status_YOLO                                              
                      count      mean      std  min  25%  50%  75%  max  
cluster                                                                  
0                     418.0  0.000000  0.00000  0.0  0.0  0.0  0.0  0.0  
1                    1795.0  0.001114  0.03337  0.0  0.0  0.0  0.0  1.0  
2                      27.0  0.000000  0.00000  0.0  0.0  0.0  0.0  0.0  

[3 rows x 304 columns]

**Examining the values and identifying the centroids**

In [77]:
centroids = Kmeans.cluster_centers_

In [78]:
print(centroids)

[[ 8.67314439e-02  9.95913149e-01  3.45771733e-05  7.71830407e-06
   2.15374262e-03  6.46117320e-04  2.37236039e-04  7.22675203e-04
   3.60488121e-04  2.60783069e-04  6.34496159e-04  8.60573750e-05
   8.16824979e-05  1.71640643e-05  1.29505982e-04  3.02613406e-04
   3.39245654e-06  2.32934060e-21  2.54109884e-21 -2.75285708e-21
   8.20563168e-22  5.58211714e-07  1.31749413e-04  4.83081180e-04
   4.68473102e-06  6.38271087e-06  6.17987836e-06  2.25196490e-05
   4.95738433e-06  3.87684842e-06 -2.48154184e-23  1.19114008e-22
   4.43358009e-06  1.73306035e-05  1.01223869e-05  1.13368738e-05
   6.93026661e-07 -1.98523347e-23]
 [ 3.61866960e-02  9.99249046e-01  8.12083513e-06  1.11308169e-05
   9.01934439e-04  5.70759101e-03  4.68189379e-04  2.92657755e-03
   6.70695642e-04  4.77779648e-04  8.31673010e-04  4.58032348e-05
   7.90928060e-05  4.86254265e-05  1.09499972e-04  9.71386736e-05
   1.38269078e-06  1.43648823e-06  1.11958230e-06  1.04919653e-06
   2.47813745e-07  1.84605967e-07  5.5172

**Most indicative variables for customer characterization**

In [79]:
description = df.groupby('cluster')['MntWines','MntFruits','MntMeatProducts','MntFishProducts','MntSweetProducts','MntGoldProds','NumDealsPurchases','NumWebPurchases','NumCatalogPurchases','NumStorePurchases','NumWebVisitsMonth','Income','AcceptedCmp1','AcceptedCmp2','AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5']
n_clients = description.size()
description = description.mean()
description['n_cients'] = n_clients


<ipython-input-79-1be30bdef527>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  description = df.groupby('cluster')['MntWines','MntFruits','MntMeatProducts','MntFishProducts','MntSweetProducts','MntGoldProds','NumDealsPurchases','NumWebPurchases','NumCatalogPurchases','NumStorePurchases','NumWebVisitsMonth','Income','AcceptedCmp1','AcceptedCmp2','AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5']


In [80]:
print(description)

           MntWines  MntFruits  MntMeatProducts  MntFishProducts  \
cluster                                                            
0         16.569378   5.418660        17.409091         8.260766   
1        375.164345  31.454039       203.151532        44.827855   
2         17.407407   7.111111        75.333333         5.111111   

         MntSweetProducts  MntGoldProds  NumDealsPurchases  NumWebPurchases  \
cluster                                                                       
0                5.839713     14.401914           1.916268         1.866029   
1               32.336490     50.896379           2.396657         4.615599   
2                5.037037     45.555556           3.888889         3.148148   

         NumCatalogPurchases  NumStorePurchases  NumWebVisitsMonth  \
cluster                                                              
0                   0.385167           2.968900           6.894737   
1                   3.209471           6.510306      

**Describing customer characteristics by clusters:**

**Cluster 0:**

- Este cluster apresenta uma média relativamente alta de gastos em vinho, carne, produtos de ouro e produtos doces.
- Os clientes deste cluster tendem a realizar um número moderado de compras com desconto, compras pela web e compras em catálogo.
- Eles também têm uma renda média relativamente alta.
- Exibem taxas moderadas de aceitação para várias campanhas de marketing.

**Cluster 1:**

- Este cluster possui valores muito baixos em todas as categorias de gastos de produtos, com exceção de gastos com ouro, que são mais altos em comparação com as outras categorias.
- Os clientes deste cluster realizam um número muito baixo de compras, tanto com desconto quanto pela web.
- A renda média deste cluster é consideravelmente mais baixa em comparação com os outros clusters.
- Eles mostram uma aceitação mínima ou nula para campanhas de marketing.

**Cluster 2:**

- Neste cluster, os gastos em todos os produtos são relativamente baixos, mas ligeiramente mais altos em comparação com o Cluster 1.
- O número médio de compras é moderado, mas inferior ao do Cluster 0.
- A renda média deste cluster é a mais baixa entre os três clusters.
- Assim como o Cluster 1, eles exibem uma aceitação mínima ou nula para campanhas de marketing.
***
***

**Cluster 0:**

- This cluster exhibits a relatively high average spending on wine, meat, gold products, and sweets.
- Customers in this cluster tend to make a moderate number of purchases with discounts, web purchases, and catalog purchases.
- They also have a relatively high average income.
- Moderate acceptance rates for various marketing campaigns are observed within this cluster.

**Cluster 1:**

- This cluster shows very low values in all product spending categories, except for spending on gold products, which are higher compared to other categories.
- Customers in this cluster make a very low number of purchases, both with discounts and online.
- The average income of this cluster is considerably lower compared to the other clusters.
- They exhibit minimal or no acceptance for marketing campaigns.

**Cluster 2:**

- In this cluster, spending on all products is relatively low, but slightly higher compared to Cluster 1.
- The average number of purchases is moderate but lower than that of Cluster 0.
- The average income of this cluster is the lowest among the three clusters.
- Similar to Cluster 1, they exhibit minimal or no acceptance for marketing campaigns.

**Based on the characteristics identified in each cluster, here are some possible strategic recommendations:**

**Cluster 0:**

1.	Campanhas Segmentadas: Investir em campanhas de marketing direcionadas a esse grupo, destacando produtos de maior interesse, como vinhos, carnes.

2.	Incentivos para Compras Online: Oferecer incentivos especiais ou descontos exclusivos para compras realizadas online, aproveitando o interesse moderado deste grupo nesse canal de compra.


3.	Programas de Fidelidade: Implementar programas de fidelidade ou recompensas para clientes recorrentes, considerando o potencial de gastos mais altos e aceitação moderada de campanhas de marketing.

**Cluster 1:**

1.	Estratégias de Engajamento: Desenvolver estratégias para aumentar o envolvimento deste grupo, oferecendo conteúdo relevante e personalizado para estimular as compras e aumentar a aceitação de campanhas de marketing.

2.	Promoções Especiais: Criar promoções especiais e descontos direcionados a esse grupo para incentivar as compras, especialmente em categorias de produtos com desempenho relativamente melhor.

**Cluster 2:**

1.	Ofertas Econômicas: Oferecer produtos e promoções com preços acessíveis para atender às preferências de compra deste grupo, que tende a gastar menos em todas as categorias de produtos.

2.	Simplificação do Processo de Compra: Simplificar o processo de compra e tornar as transações mais convenientes, especialmente em lojas físicas, para incentivar o aumento das compras.


3.	Educação sobre Produtos: Fornecer informações e educação sobre os produtos oferecidos, destacando os benefícios e características únicas para atrair o interesse deste grupo com menor engajamento.

***
***

**Cluster 0:**

1.	Segmented Campaigns: Invest in targeted marketing campaigns aimed at this group, highlighting high-interest products such as wines and meats.

2.	Incentives for Online Purchases: Offer special incentives or exclusive discounts for purchases made online, leveraging this group's moderate interest in this purchasing channel.

3.	Loyalty Programs: Implement loyalty programs or rewards for repeat customers, considering the potential for higher spending and moderate acceptance of marketing campaigns.

**Cluster 1:**

1.	Engagement Strategies: Develop strategies to increase engagement within this group by offering relevant and personalized content to stimulate purchases and increase acceptance of marketing campaigns.

2.	Special Promotions: Create special promotions and discounts targeted at this group to encourage purchases, especially in categories with relatively better performance.

**Cluster 2:**

1.	Economical Offers: Provide affordable products and promotions to meet the purchasing preferences of this group, which tends to spend less across all product categories.

2.	Simplified Purchase Process: Simplify the purchase process and make transactions more convenient, especially in physical stores, to encourage increased purchases.

3.	Product Education: Provide information and education about the products offered, highlighting benefits and unique features to attract the interest of this group with lower engagement.
